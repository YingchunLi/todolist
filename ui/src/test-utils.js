import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';

import configureStore from './store';

import {Provider} from 'react-redux';

// wrapper for store
export const withStore = (WrappedComponent, state = {}) => (props) => {
  const store = configureStore(state);
  return (
    <Router>
      <Provider store={store}>
        <WrappedComponent {...props} />
      </Provider>
    </Router>
  );
};

