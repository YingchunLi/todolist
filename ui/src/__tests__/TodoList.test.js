import React from "react";
import {MemoryRouter, Route} from "react-router-dom";

import renderer from "react-test-renderer";
import {shallow, mount} from "enzyme";

import {withStore} from '../test-utils';
import ConnectedTodoList, {TodoList} from "../TodoList";

describe("TodoList", () => {
    it("maps to /todos", () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={[`/todos`]}>
          <switch>
            <Route path="/todos" component={TodoList}/>
          </switch>
        </MemoryRouter>
      );

      expect(wrapper.find(TodoList).exists()).toBe(true);
      expect(wrapper.find(TodoList).props().location.pathname).toBe("/todos");
    });

    it("has a title and a table", () => {
      const wrapper = shallow(
        <TodoList dispatch={(action) => {
        }}/>
      );

      expect(wrapper.find('h2').length).toBe(1);
      expect(wrapper.find('h2').text()).toBe('This is the todo list page');
      expect(wrapper.find('table').length).toBe(1);
    });

    it("has no todos", () => {
      const component = renderer.create(
        <TodoList/>
      ).toJSON();
      expect(component).toMatchSnapshot();
    });

    it("has some todos", () => {
      const todos = [
        {
          id: 1,
          name: 'milk',
          description: 'remember the milk',
          dueDate: new Date(2018, 12, 31).toISOString(),
          status: 'Pending'
        },
      ];
      const component = renderer.create(
        <MemoryRouter initialEntries={[`/todos`]}>
          <TodoList todos={todos}/>
        </MemoryRouter>

      ).toJSON();
      expect(component).toMatchSnapshot();
    });

    it("connected component works", () => {
      const todos = [
        {
          id: 1,
          name: 'milk',
          description: 'remember the milk',
          dueDate: new Date(2018, 12, 31).toISOString(),
          status: 'Pending'
        },
      ];
      const TodoListWithStore = withStore(ConnectedTodoList);
      const component = renderer.create(
        <TodoListWithStore />
      ).toJSON();
      expect(component).toMatchSnapshot();
    });

  }
);
