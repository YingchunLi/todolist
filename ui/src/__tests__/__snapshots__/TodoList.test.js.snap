// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList connected component works 1`] = `
<div>
  <h2>
    This is the todo list page
  </h2>
  <table
    border="1"
  >
    <thead>
      <tr>
        <th>
          id
        </th>
        <th>
          name
        </th>
        <th>
          description
        </th>
        <th>
          due date
        </th>
        <th>
          status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`TodoList has no todos 1`] = `
<div>
  <h2>
    This is the todo list page
  </h2>
  <table
    border="1"
  >
    <thead>
      <tr>
        <th>
          id
        </th>
        <th>
          name
        </th>
        <th>
          description
        </th>
        <th>
          due date
        </th>
        <th>
          status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`TodoList has some todos 1`] = `
<div>
  <h2>
    This is the todo list page
  </h2>
  <table
    border="1"
  >
    <thead>
      <tr>
        <th>
          id
        </th>
        <th>
          name
        </th>
        <th>
          description
        </th>
        <th>
          due date
        </th>
        <th>
          status
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="/todos/1"
            onClick={[Function]}
          >
            1
          </a>
        </td>
        <td>
          milk
        </td>
        <td>
          remember the milk
        </td>
        <td>
          2019-01-30T14:00:00.000Z
        </td>
        <td>
          Pending
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
